flamedisx/x1t_sr1.py

```diff
@@ -15,21 +19,126 @@
+
+def_field = 81.
+
+def_recon_pivot = 0.49
 
 ##
-# Flamedisx sources
+# Loading Pax reconstruction bias
+##
+dummy_base = os.path.dirname(os.path.realpath(__file__)) + '/dummy_maps/'
+
+path_recon_bias_mean_s1 = ['/home/peaelle42/software/bbf/bbf/data/ReconstructionS1BiasMeanLowers_SR1_v2.json',
+         '/home/peaelle42/software/bbf/bbf/data/ReconstructionS1BiasMeanUppers_SR1_v2.json']
+path_recon_bias_mean_s2 = ['/home/peaelle42/software/bbf/bbf/data/ReconstructionS2BiasMeanLowers_SR1_v2.json',
+         '/home/peaelle42/software/bbf/bbf/data/ReconstructionS2BiasMeanUppers_SR1_v2.json']
+
+path_dummy_zeros_s1 = ['dummy_zeros_s1.json', 'dummy_zeros_s1.json']
+path_dummy_zeros_s1 = [dummy_base+x for x in path_dummy_zeros_s1]
+path_dummy_zeros_s2 = ['dummy_zeros_s2.json', 'dummy_zeros_s2.json']
+path_dummy_zeros_s2 = [dummy_base+x for x in path_dummy_zeros_s2]
+
+def read_bias_tf(path_bag):
+    # Achtung: 
+    # Fundamentally assumes upper and lower bounds have exactly the same support definition
+    data_bag = []
+    yy_ref_bag = []
+    for loc_path in path_bag:
+        with open(loc_path) as json_file:
+            tmp = json.load(json_file)
+            yy_ref_bag.append(tf.convert_to_tensor(tmp['map'], dtype=fd.float_type()))
+            data_bag.append(tmp)
+    support_def = tmp['coordinate_system'][0][1] 
+    
+    return yy_ref_bag, support_def
+
+def cal_bias_tf(sig, fmap, support_def, pivot_pt):
+    tmp = tf.convert_to_tensor(sig, dtype=fd.float_type())
+    bias_low = tfp.math.interp_regular_1d_grid(x=tmp, 
+            x_ref_min=support_def[0], x_ref_max=support_def[1], y_ref=fmap[0],
+            fill_value='constant_extension')
+    bias_high = tfp.math.interp_regular_1d_grid(x=tmp, 
+            x_ref_min=support_def[0], x_ref_max=support_def[1], y_ref=fmap[1],
+            fill_value='constant_extension')
+
+    tmp = tf.math.subtract(bias_high, bias_low)
+    tmp1 = tf.math.scalar_mul(pivot_pt, tmp)
+    bias_out = tf.math.add(tmp1, bias_low)
+    bias_out = tf.math.add(bias_out, tf.ones_like(bias_out))
+
+    return bias_out
+
 ##
+# Loading combined cuts acceptances
+##
+def itp_cut_accept_tf(sig, fmap, support_def):
+    accept_out = tf.squeeze(tfp.math.interp_regular_1d_grid(x=sig, 
+            x_ref_min=support_def[0], x_ref_max=support_def[1], y_ref=fmap,
+            fill_value='constant_extension'))
+    return accept_out
+
+# Defining paths
+path_cut_accept_s1 = ['/home/peaelle42/software/bbf/bbf/data/S1AcceptanceSR1_v7_Median.json']
+path_cut_accept_s2 = ['/home/peaelle42/software/bbf/bbf/data/S2AcceptanceSR1_v7_Median.json']
 
+path_dummy_ones_s1 = [dummy_base+'dummy_ones_s1.json']
+path_dummy_ones_s2 = [dummy_base+'dummy_ones_s2.json']
 
+##
+# Flamedisx sources
+##
 class SR1Source:
-    drift_velocity = 1.335 * 1e-4   # cm/ns
+    drift_velocity = def_drift_vel
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        try:
```

Comment:
How about just loading the dummy maps in this PR, and leave the bbf map loading to after we've merged with master? Then you don't need this duplicated code. 

A catch-all try-except without re-raise is dangerous in general (e.g. it also catches SyntaxError; maybe the code just never runs), though I understand this code is just a placeholder and the print statements tell you which path you took. You don't want 10+ print statements in a production code that loads several sources though.